### PR TITLE
RDKBWIFI-12 Fix for using network order byte of em_tlv_type_profile.

### PR DESCRIPTION
**Pull request by marcin-matula-2329518**
It looks like tlv->len is already converted into network bytes order (htons). Using this member in host architecture seems require converting it back into host bytes order (ntohs). Without that could be observed undefined behaviour (in it case memory corruption and segfault).

### CODE REVIEW

File: src/em/config/em_configuration.cpp [hunk 1]
```diff
@@ -1021,7 +1021,7 @@ int em_configuration_t::handle_topology_response(unsigned char *buff, unsigned i
 
         } else {
             found_profile = true;
-			memcpy(&profile, tlv->value, tlv->len);
+			memcpy(&profile, tlv->value, ntohs(tlv->len));
```
Comment (marcin-matula-2329518) [inline src/em/config/em_configuration.cpp:1024]: It seems be quick fix. Unfortunately in the code I see to tlv->len is set variable converted by using htons function

Comment (marcin-matula-2329518) [reply src/em/config/em_configuration.cpp:1024]: link to jira with more detialed description https://jira.rdkcentral.com/jira/browse/RDKBWIFI-12

Comment (marcin-matula-2329518) [reply src/em/config/em_configuration.cpp:1024]: Not to merge yet. To verify one thing from my side.

**Review by soumyasmunshi — APPROVED** [review]

**Comment by marcin-matula-2329518** [comment]
Converted to draft, to verify if tlv->value copied into profile are valid data.

**Comment by marcin-matula-2329518** [comment]
Tests:

Sent tlv:
Thread 3 "onewifi_em_ctrl" hit Breakpoint 3, em_configuration_t::send_topology_query_msg (this=0x5555629d30) at /home/mmatula/easymesh_project/unified-wifi-mesh/build/ctrl/../..//src/em/config/em_configuration.cpp:319
319         tmp += (sizeof(em_tlv_t) + sizeof(em_enum_type_t));
$30 = {type = 179 '\263', len = 256, value = 0x7ff6f973b1 "\003"}
$31 = 1 (tlv length in host bytes order)
$32 = 256 (tlv length in network bytes order)
$33 = 0x7ff6f973b1 "\003" (tlv value)

Received tlv:
Thread 3 "onewifi_em_ctrl" hit Breakpoint 5, em_configuration_t::handle_topology_response (this=0x5555629d30, buff=0x7ff0029510 "\270'\353\020\031\214\270'\353\020\031\214\211:", len=200) at /home/mmatula/easymesh_project/unified-wifi-mesh/build/ctrl/../..//src/em/config/em_configuration.cpp:1024
1024                            memcpy(&profile, tlv->value, ntohs(tlv->len));
$41 = {type = 179 '\263', len = 256, value = 0x7ff002956a "\003\267"}
$42 = 1 (tlv length in host bytes order)
$43 = 256 (tlv length in network bytes order)
$44 = 0x7ff002956a "\003\267" (tlv value, in the case if tlv length in host bytes order is 1 it is just "\003")

**Pull request merged**